### PR TITLE
ADD: Mac client kind cluster hack script

### PR DESCRIPTION
**Pull request by danpopnyc**
edited default script to work on a mac client (disabled swapoff, and mapping to in memory etcd tmpfs

### CODE REVIEW

File: hack/setup-mac-kind.sh [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+#sudo swapoff -a
+#sudo rm -f /swapfile
+
+# Use in-memory storage to avoid etcd server timeouts.
+# https://kubernetes.slack.com/archives/CEKK1KTN2/p1615134111016300
+# https://github.com/kubernetes-sigs/kind/issues/845
+sudo mkdir -p /tmp/etcd
+#sudo mount -t tmpfs tmpfs /tmp/etcd
+
+curl -Lo ./kind "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64"
```
Comment (vaikas) [inline hack/setup-mac-kind.sh:87]: I was testing with arm64 when I was doing local testing, maybe change to arm64?

Comment (vaikas) [reply hack/setup-mac-kind.sh:87]: or maybe change this to:
kind-$(uname)-$(uname -m)

Comment (danpopnyc) [reply hack/setup-mac-kind.sh:87]: teh things i commented out related to the mapped drive to tmpfs which there is no concept of in MacOS i.e. https://superuser.com/questions/448585/is-there-an-implementation-of-tmpfs-for-mac-os-x

i removed the tmpfs mount and the mounted drives for the kind nodes

Comment (vaikas) [reply hack/setup-mac-kind.sh:87]: Yeah, I had to make those changes as well :) However, is your mac intel or arm based?

Comment (danpopnyc) [reply hack/setup-mac-kind.sh:87]: m1 (apple silcon)

Comment (vaikas) [reply hack/setup-mac-kind.sh:87]: ok, well, if it worked for you as is, let's get it in, I'll retry it on my setup :)

**Pull request merged**